### PR TITLE
Merge nRF and nRF Connect Dockerfiles

### LINKED ISSUE #1549: Merge "nRF" and "nRF Connect" Dockerfiles
#### Problem
As @rwalker-apple pointed out in #1529 Dockerfiles for _nrf_ and _nrfconnect_ platforms share some dependencies, so the scripts might be merged to avoid maintenance burden related to the need to update them both in the future.

 #### Proposed Solution
* _nrf_ and _nrfconnect_ Dockerfiles merged into a single multi-stage Dockerfile
* change in _integrations/docker/build.sh_ to support selection of a target from a multi-stage Dockerfile.

I will take care to implement the solution.

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
As @rwalker-apple pointed out in #1529 Dockerfiles for nrf and nrfconnect platforms share some dependencies, so the scripts might be merged to avoid maintenance burden related to the need to update them both in the future.

 #### Summary of Changes
* nrf and nrfconnect Dockerfiles merged into a single multi-stage Dockerfile
* change in integrations/docker/build.sh to support selection of a target from a multi-stage Dockerfile.

fixes #1549

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Are the images sufficiently different to warrant separate  docker images?

I originally understood the merge request as 'build single dockerfile for all nRF development'. This PRs generates the same output except that it uses a single dockerfile ... for me updating docker seems to have become more complex as a result instead of easier.

**Comment by Damian-Nordic** [comment]
> Are the images sufficiently different to warrant separate docker images?
> 
> I originally understood the merge request as 'build single dockerfile for all nRF development'. This PRs generates the same output except that it uses a single dockerfile ... for me updating docker seems to have become more complex as a result instead of easier.

Hmm... The reason I thought it would be better to have separate images is that nRF Connect SDK adds over 1GB of files which are irrelevant for someone who wants to use nRF5 SDK. And similarly the other way around (although nRF5 SDK adds a bit less). I understood that the merge was requested to avoid the need to update two Dockerfiles instead of one when it's necessary to update toolchain etc. and I couldn't find an easier way to meet those two requirements other than using a multi-stage dockerfile (since Docker doesn't support any "include other Dockerfile" functionality).

If it's still controversial I can create a single nRF docker image and if we decide we can make developers' lives easier by publishing a minimal nRF Connect SDK image we can probably maintain it on Nordic side. Let me know if that's ok.

File: integrations/docker/images/chip-build-nrf-platform/Dockerfile [hunk 1]
```diff
@@ -30,6 +24,49 @@ RUN set -x \
     | tar jxvf -) \
     && : # last line
 
+# ==================================================
+# nRF5 SDK image (target: chip-build-nrf-platform)
+# ==================================================
+FROM common AS chip-build-nrf-platform
```
Comment (rwalker-apple) [inline integrations/docker/images/chip-build-nrf-platform/Dockerfile:30]: I'm new to this syntax.  Where does this "FROM" directive end?  At the next "FROM"?

File: integrations/docker/images/chip-build-nrfconnect-platform/run.sh [hunk 2]
```diff
@@ -1 +0,0 @@
-../../run.sh
```
Comment (rwalker-apple) [inline integrations/docker/images/chip-build-nrfconnect-platform/run.sh:1]: deleting this file and directory means that there's now no facility to do a "run" in the nrf-connect image.

**Comment by rwalker-apple** [comment]
This isn't the way I was expecting this to come out.  I would have been happier with an image that has both SDKs and exports all the env vars (they don't conflict).

@Damian-Nordic if you feel strongly that these environments should be separate, I'd prefer that you refactored according to the pattern already in place here:
* construct a base image directory (with a run.sh, build.sh, and Dockerfile) that contains the common deb packages
* construct an image directory for each of the environments that FROM the base image directory

**Review by rwalker-apple — CHANGES_REQUESTED** [review]
This isn't the way I was expecting this to come out.  I would have been happier with an image that has both SDKs and exports all the env vars (they don't conflict).

@Damian-Nordic if you feel strongly that these environments should be separate, I'd prefer that you refactored according to the pattern already in place here:
* construct a base image directory (with a run.sh, build.sh, and Dockerfile) that contains the common deb packages
* construct an image directory for each of the environments that FROM the base image directory

File: integrations/docker/images/chip-build-nrf-platform/Dockerfile (hunk 1 above)
Comment (Damian-Nordic) [reply integrations/docker/images/chip-build-nrf-platform/Dockerfile:30]: yes

**Comment by Damian-Nordic** [comment]
> This isn't the way I was expecting this to come out. I would have been happier with an image that has both SDKs and exports all the env vars (they don't conflict).

As I mentioned before I just wanted the images to be smaller so that they would be faster to setup by users, but maybe that's not important, so I'll do what you propose.

**Review by BroderickCarlin — APPROVED** [review]

**Comment by rwalker-apple** [comment]
> > This isn't the way I was expecting this to come out. I would have been happier with an image that has both SDKs and exports all the env vars (they don't conflict).
> 
> As I mentioned before I just wanted the images to be smaller so that they would be faster to setup by users, but maybe that's not important, so I'll do what you propose.

the proposal above of a "base" image with the common bits achieve the smaller images goal

**Review by mspang — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
> the proposal above of a "base" image with the common bits achieve the smaller images goal

yes, I understand and thank you for that, but after second thought we should be fine with merging both nRF images and if the size becomes an issue we'll return to the proposal.

**Review by rwalker-apple — APPROVED** [review]
lgtm

**Review by woody-apple — APPROVED** [review]

**Pull request merged**